benchmarks/benchmark_serving.py

```diff
@@ -448,6 +449,16 @@ async def benchmark(
 
     pbar = None if disable_tqdm else tqdm(total=len(input_requests))
 
+    semaphore = asyncio.Semaphore(max_concurrency) if max_concurrency else None
+
+    async def limited_request_func(request_func_input, pbar):
+        if semaphore:
+            async with semaphore:
```

Comment:
Is there an asyncio version of nullcontext so that we can unify this branch?

Reply:
I think these can co-exist, but the scenario we're trying to benchmark on is:

- The inference server is set up for server X users, with the assumption that each use will not send async calls
- Therefore the max concurrency/number of clients "in conntection" with the inference server is X
- Note, we can still have QPS > X, since it's possible that a single user can make 2 consequent calls within one second if each call is really short.

Hope that makes sense!

Reply:
No I just meant the coding style. Was thinking about something like this

```python
from contextlib import nullcontext

# nullcontext can be used with sync or async since Python 3.10.
semaphore = asyncio.Semaphore(max_concurrency) if max_concurrency else nullcontext()
async with semaphore:
    return await request_func(
        request_func_input=request_func_input, pbar=pbar)
```

Reply:
ah yes, forgot to address this, sorry

Reply:
nullcontext appears to work fine here. Thanks for the suggestion!